harvester_e2e_tests/integration/test_vm.py

```diff
@@ -0,0 +1,327 @@
+                is_ready = True
+                break
+
+        if is_ready and not node_info.get('spec', {}).get('unschedulable', False):
+            node_names.append(node_info['metadata']['name'])
+
+    assert 2 <= len(node_names), (
+        f"The cluster only have {len(node_names)} available node. It's not enough."
+    )
+    yield node_names
+
+
+def get_vm_agent_connected_cnt(api_client, vm_names):
+    cnt = 0
+    for vm_name in vm_names:
+        _, data = api_client.vms.get(vm_name)
+        if data.get('status', {}).get('ready', False):
+            _, data = api_client.vms.get_status(vm_name)
+            if data['status']['conditions'][-1]['status'] == 'True':
+                cnt += 1
+    return cnt
+
+
+@pytest.mark.p0
+@pytest.mark.virtualmachines
+def test_multiple_migrations(api_client, unique_name, focal_image, wait_timeout,
+                             available_node_names):
+    vm_names = [f"migrate-1-{unique_name}", f"migrate-2-{unique_name}"]
+
+    vm_spec = api_client.vms.Spec(1, 1)
+    vm_spec.add_image('disk-0', focal_image['id'])
+    vm_data = []
+    for vm_name in vm_names:
+        code, data = api_client.vms.create(vm_name, vm_spec)
+        assert 201 == code, (
+            f"Failed to create VM {vm_name} with error: {code}, {data}"
+        )
+        vm_data.append(data)
+
+    for vm_name in vm_names:
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        code, data = None, None
+        while endtime > datetime.now():
+            code, data = api_client.vms.get(vm_name)
+            if data.get('status', {}).get('ready', False):
+                code, data = api_client.vms.get_status(vm_name)
+                if data['status']['conditions'][-1]['status'] == 'True':
+                    break
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"Can't find VM {vm_name} with {wait_timeout} timed out\n"
+                f"Got error: {code}, {data}"
+            )
+
+    vmi_data = []
+    for vm_name in vm_names:
+        code, data = api_client.vms.get_status(vm_name)
+        assert 200 == code, (
+            f"Failed to get VM {vm_name} with error: {code}, {data}"
+        )
+        vmi_data.append(data)
+
+    vm_src_dst_hosts = {}  # {vm_name: [src_host, dst_host]}
+    for vm_name, vm_datum in zip(vm_names, vmi_data):
+        src_host = vm_datum['status']['nodeName']
+        dst_host = next(n for n in available_node_names if n != src_host)
+        vm_src_dst_hosts[vm_name] = [src_host, dst_host]
+
+        code, data = api_client.vms.migrate(vm_name, dst_host)
+        assert code == 204, (
+            f"Can't migrate VM {vm_name} from host {src_host} to {dst_host}",
+            f"Got error: {code}, {data}"
+        )
+
+    for vm_name, src_dst_hosts in vm_src_dst_hosts.items():
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        code, data = None, None
+        while endtime > datetime.now():
```

Comment:
same as above, we would need use `while-for`, if timed out, we would know all VMs' status. (when raising exception)